R/aggregate.R

```diff
@@ -114,50 +108,19 @@ agg_climate_data <- function(year, data_raster, climate_var, daily_agg, trans =
     clim_daily <- clim_daily - 273.15
   }
 
-  ## Nonlinearities
-  ## -----------------------------------------------
+  ## convert prcp m to mm
+  if(climate_var == 'prcp'){
 
+    clim_daily <- clim_daily * 1000
+  }
 
-  # Polynomial
-  if(trans == 'polynomial'){
-
-    k <- trans_specs
-    poly_orders <- seq(1:k) # Compute values from 1 to K
-    list_length <- length(poly_orders) # How many lists are in the final object
-    list_names <- sapply(1:list_length, FUN=function(x){paste("order", poly_orders[x], sep="_")})
-
-    # For each daily layer, raise the value to k, k-1, k-2 etc. until 1
-    r <- lapply(poly_orders, FUN=function(x){clim_daily ^ x})
-
-    ## Function: Set names of data table by month, change from wide to long format, rename based on polynomial orders
-    create_dt <- function(x){
-
-      # Should output raster cells x/y with 365 days as column names
-      dt <- as.data.table.raster(r[[x]], xy=TRUE)
-
-      # Set column names with months
-      new_names <- c('x', 'y', layer_names)
-      data.table::setnames(dt, new_names)
-
-      # Change from wide to long format
-      dt = data.table::melt(dt, id.vars = c("x", "y"))
-
-      # Update variable names
-      var_names <- c('date', list_names[x])
-      data.table::setnames(dt, old=c('variable', 'value'), new=var_names)
-    }
-
-    # Make each raster layer a data.table
-    list_dt <- lapply(1:list_length, create_dt)
+  # Return a list of the necessary objects
+  return(list(clim_daily, weights_dt, layer_names))
+}
 
-    # Merge all data tables together
-    clim_dt <- list_dt[[1]]
-    for(i in 2:list_length){
-      dt_m <- list_dt[[i]]
-      clim_dt <- merge(clim_dt, dt_m, by=c('x', 'y', 'date'))
-    }
 
-  }
+# Function to merge with geoweights and aggregate by polygon
+after_trans <- function(clim_dt, weights_dt, list_names, second_weights, agg_to){
```

Comment:
@tliddell4 i'm missing something...where did the nonlinear transformations go?? we are already at the "after trans" function but i don't see the nonlinearities? if this is just a function being placed wherever bc you don't need it to go in any particular order then i guess it's ok?